Library/Homebrew/sandbox.rb

```diff
@@ -68,6 +69,11 @@ def allow_write_cellar(formula)
     allow_write_path formula.var
   end
 
+  # Xcode projects expect access to certain cache/archive dirs.
+  def allow_write_xcode
+    allow_write_path "/Users/#{ENV["USER"]}/Library/Developer/Xcode/DerivedData/"
```

Comment:
Oh, that's a neat idea. It wouldn't catch everything but it'd reduce an awful lot of the formulae that do wander. Will check it out.